aea/aea.py

```diff
@@ -137,13 +134,15 @@ def handle(self, envelope: Envelope) -> None:
                 default_handler.send_invalid_message(envelope)
             return
 
-        handler = self.resources.handler_registry.fetch(protocol.id)
-        if handler is None:
+        handlers = self.resources.handler_registry.fetch(protocol.id)
+        if handlers is None:
             if default_handler is not None:
                 default_handler.send_unsupported_skill(envelope, protocol)
             return
 
-        handler.handle_envelope(envelope)
+        # each handler independently acts on the message
```

Comment:
@marcofavorito this is needed for now so we can add several skills for a protocol. Later the orchestrator/director can be introduced.